projects/igniteui-angular/src/lib/grids/filtering/grid-filtering-cell.component.ts

```diff
@@ -309,4 +313,35 @@ export class IgxGridFilteringCellComponent implements AfterViewInit, OnInit {
             this.cdr.detectChanges();
         }
     }
+
+    private ScrollToChip(columnIndex: number, scrollRight: boolean) {
+        this.filteringService.grid.nativeElement.focus({preventScroll: true});
+        this.filteringService.columnToFocus = this.filteringService.grid.visibleColumns[columnIndex];
```

Comment:
Setting `columnToFocus` and `focusNext` happens only in filtering cell component and in navigation service. And in the both places these two fields are set together and are never read. So in order to make the code a little bit easier to use and error prone we could create a method in the filtering service to set those two fields. They are only used in the service so make them private.

Reply:
focusNext is read in the filtering cell when we decide in which direction we should move the focus.